plugins/microservices/msi_json/json_value.cpp

```diff
@@ -0,0 +1,141 @@
+            }
+            else if (json_value.is_number_float()) {
+                const auto v = json_value.get<json::number_float_t>();
+                rodsLog(LOG_DEBUG, "JSON value = [%f]", v);
+                fillStrInMsParam(_value, std::to_string(v).c_str());
+            }
+            else if (json_value.is_boolean()) {
+                const auto v = json_value.get<bool>();
+                rodsLog(LOG_DEBUG, "JSON value = [%i]", v);
+                fillStrInMsParam(_value, v ? "true" : "false");
+            }
+            else if (json_value.is_null()) {
+                rodsLog(LOG_DEBUG, "JSON value = [null]");
+                fillStrInMsParam(_value, "null");
+            }
+            else if (json_value.is_array() || json_value.is_object()) {
+                fillStrInMsParam(_value, json_value.type_name());
+            }
+            else {
+                // This is only reachable if the JSON standard defines new data types.
+                // Returning this error code helps to future-proof the iRODS server by providing signal
+                // to the developer or administrator that this microservice or the library needs to updated.
+                return TYPE_NOT_SUPPORTED;
+            }
+        }
+        catch (const irods::exception& e) {
+            rodsLog(LOG_ERROR, "Caught exception while fetching JSON value. [%s]", e.client_display_what());
+            return static_cast<int>(e.code());
+        }
+        catch (const boost::bad_any_cast& e) {
+            rodsLog(LOG_ERROR, "Caught exception while fetching JSON value. [%s]", e.what());
+            return INVALID_ANY_CAST;
+        }
+        catch (const std::exception& e) {
+            rodsLog(LOG_ERROR, "Caught exception while fetching JSON value. [%s]", e.what());
+            return SYS_LIBRARY_ERROR;
+        }
+
+        return 0;
+    } // msi_impl
+
+    template <typename... Args, typename Function>
+    auto make_msi(const std::string& name, Function func) -> irods::ms_table_entry*
+    {
+        auto* msi = new irods::ms_table_entry{sizeof...(Args)}; // NOLINT(cppcoreguidelines-owning-memory)
+        msi->add_operation(name, std::function<int(Args..., ruleExecInfo_t*)>(func));
+        return msi;
+    } // make_msi
+} // anonymous namespace
+
+extern "C" auto plugin_factory() -> irods::ms_table_entry*
+{
+    return make_msi<MsParam*, MsParam*, MsParam*>("msi_json_value", msi_impl);
+} // plugin_factory
+
+#ifdef IRODS_FOR_DOXYGEN
+/// \brief Fetches a value within a JSON structure.
+///
+/// \param[in]     _json_handle  The handle of a JSON structure obtained via msi_json_parse().
+/// \param[in]     _json_pointer \parblock
+/// A JSON pointer to the element of interest. If the element cannot be represented in the iRODS
+/// Rule Language, its type will be written to \p _value instead.
+///
+/// The following type strings can be returned:
+/// - array
+/// - binary
```

Comment:
Remove this line.

The JSON lib (v3.7.3) does not support binary data.

Reply:
maybe even better to say that binary is explicitly not supported, for that reason.  removes any doubt in the reader.

Reply:
Done.